src/bcWithoutPred+Pred.stan

```diff
@@ -0,0 +1,226 @@
+functions {
+  vector gp_pred_rng(int n, int N, int n_pred, int p, int q, vector z, 
+                     row_vector[] xf, row_vector[] xc, row_vector[] x_pred,
+                     row_vector tf, row_vector[] tc,
+                     row_vector beta_eta, real lambda_eta,
+                     row_vector beta_delta, real lambda_delta,
+                     real sigma) {
+    vector[n_pred] y_pred;
+    {
+      matrix[N,N] sigma_eta;  //sigma_eta + sigma_delta = sigma_z
+      matrix[n,n] sigma_delta;  //sigma_eta + sigma_delta = sigma_z
+      matrix[N,N] sigma_z;  //sigma_z = s11
+      matrix[N,N] L_K;  //L_K'*L_K = s11
+      vector[N] L_K_div_y;
+      vector[N] K_div_y;  //K_div_y = s11{-1} y
+      matrix[N, n_pred] k_x1_x2;  //k_x1_x2 = s12
+      vector[n_pred] y_pred_mu;  //y_pred_mu = E{y(x_pred)|y,.} = s21 s11{-1} y
+      matrix[N,n_pred] v_pred;  //v_pred' * v_pred = s21 s11{-1} s12
+      matrix[n_pred,n_pred] sigma_eta_pred; //sigma_eta_pred + sigma_delta_pred = s22
+      matrix[n_pred,n_pred] sigma_delta_pred;  //sigma_eta_pred + sigma_delta_pred = s22
+      matrix[n_pred,n_pred] cov_y_pred;  //cov_y_pred = COV{y(x_pred, tf)|y,.} = s22 - s21 s11{-1} s12
+      row_vector[p+q] xt[N];
+      row_vector[p+q] xt_pred[n_pred];
+      
+      // xt = [[xt,tf],[xc,tc]]
+      for (i in 1:n) {
+        xt[i] = append_col(xf[i],tf);
+      }
+      for (i in (n+1):N) {
+        xt[i] = append_col(xc[i-n],tc[i-n]);
+      }
+      for (i in 1:n_pred) {
+        xt_pred[i] = append_col(x_pred[i],tf);
+      }
+
+	  // Evaluate s11
+      for (i in 1:(N-1)) {
+        sigma_eta[i, i] = 1/lambda_eta + sigma;
+        for (j in (i+1):N) {
+          sigma_eta[i, j] = exp(-dot_self((xt[i] - xt[j]) .* beta_eta))/lambda_eta;
+          sigma_eta[j, i] = sigma_eta[i, j];
+        }
+      }
+      sigma_eta[N, N] = 1/lambda_eta + sigma;
+
+      // elements of sigma_delta and add observation errors
+      for (i in 1:(n-1)) {
+        sigma_delta[i, i] = 1/lambda_delta + 1e-9;
```

Comment:
set delta as a variable in the transformed data block instead of adding 1e-9 at every line